### PR TITLE
Fix #68

### LINKED ISSUE #68: Certain Flow errors are ignored
Some of our errors are being ignored. It only seems to occur when the error concerns two different files, e.g. when incorrect props are passed to a React component that was imported from another file.

After some debugging, it seems to be filtered by this filepath equality conditional: https://github.com/amilajack/eslint-plugin-flowtype-errors/blob/06138df15300f525e61670af68b1153eee624e80/src/collect.js#L140

Here is the `flow status` output of an error that is be swallowed by the plugin. Note that each message array includes differing filenames, hence the issue.

```
src/components/conversation-wrap.js:14
 14:     ? <ReassignmentPage {...props} />
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ React element `ReassignmentPage`
 17: class ReassignmentPage extends React.Component<void, PropsT, StateT> {
                                                          ^^^^^^ property `isLoading`. Property not found in. See: src/handlers/reassignment-page.js:17
 14:     ? <ReassignmentPage {...props} />
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ props of React element `ReassignmentPage`

src/handlers/reassignment-page.js:17
 17: class ReassignmentPage extends React.Component<void, PropsT, StateT> {
                                                          ^^^^^^ property `isLoading`. Property not found in
 12: const ConversationWrap = (props: PropsT): React.Element<ConversationWrap> =>
                                                             ^^^^^^^^^^^^^^^^ prototype. See: src/components/conversation-wrap.js:12
```

Happy to provide more debugging info if needed.

### PR DESCRIPTION
**Pull request by jdmota**
Fixes #68.

I also manually added the tests from #74.

### CODE REVIEW

**Comment by jdmota** [comment]
On Appveyor, it seems that no tests are run. Not sure why.

On Travis (and on my Windows 10), and the tests pass.

File: src/collect.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ try {
   /* eslint-enable */
 }
 
+function mainLocOfError(error) {
```
Comment (amilajack) [inline src/collect.js]: Would be nice if you could annotate type types of this function:

```js
type errorObject = {
  operation?: {
    loc
  },
  message: Array<idkwhatThisTypeis>
}

function mainLocOfError(error: errorObject): string {
  const { operation, message } = error;
  return operation && operation.loc || message[0].loc;
}
```

**Comment by amilajack** [comment]
@jdmota Thanks for the fix!  🎉 🎉 🎉

**Comment by amilajack** [comment]
Pushing a prepatch at `3.0.3-0` would be great if you could test it out

**Comment by jdmota** [comment]
It seems to be working 😄 

Maybe in the future, the eslint rule should be tested instead of just testing the `collect` function, so we don't need to manually test it everytime 😜

**Comment by amilajack** [comment]
I've found some weird edge cases where sometimes certain flow error object are not present and we end up with 'undefined is not a function', which caused the plugin to fail. Would be nice if we could have some automated tests that clone flow codebases and run the plugin on them

**Comment by lydell** [comment]
Seems to be working for me as well. Thanks!
